### PR TITLE
[XPTIFW] Enable in-tree builds

### PR DESCRIPTION
**Pull request by alexbatashev**
This time Pull Request also includes reasonable fixes for warnings and disables targets, that are currently not buildable.

### CODE REVIEW

File: xptifw/CMakeLists.txt [hunk 1]
```diff
@@ -14,17 +17,28 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Build type (default Release)" FORCE)
 endif()
 
-project (xptifw)
+foreach(flag_var
```
Comment (bader) [inline xptifw/CMakeLists.txt:20]: Please, add a comment with the reason to make this replacement.

File: xptifw/CMakeLists.txt [hunk 2]
```diff
@@ -14,17 +17,28 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
...
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
+endif()
 
 set(CMAKE_BINARY_DIR ${CMAKE_SOURCE_DIR}/lib/${CMAKE_BUILD_TYPE})
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR})
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR})
 
 include_directories(${CMAKE_SOURCE_DIR}/include ${XPTI_DIR}/include)
 add_subdirectory(src)
+# TODO enable samples and tests back once build problems are resolved.
```
Comment (bader) [inline xptifw/CMakeLists.txt:36]: Could you elaborate more on build problems, please? What problems should be resolved to add tests and samples?

File: xptifw/CMakeLists.txt (hunk 1 above)
Comment (alexbatashev) [reply xptifw/CMakeLists.txt]: Actually, all the public interfaces are wrapped with `extern C`, so this is not needed. Removed.

File: xptifw/CMakeLists.txt (hunk 2 above)
Comment (alexbatashev) [reply xptifw/CMakeLists.txt:30]: Added a comment

**Comment by alexbatashev** [comment]
@intel/llvm-reviewers-runtime @tovinkere @andykaylor friendly ping

**Review by tovinkere — APPROVED** [review]
LGTM

**Review by andykaylor — APPROVED** [review]
lgtm

**Comment by romanovvlad** [comment]
@smaslov-intel  Could you please review the patch?

File: xptifw/CMakeLists.txt [hunk 3]
```diff
@@ -14,17 +17,25 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Build type (default Release)" FORCE)
 endif()
 
-project (xptifw)
+if(MSVC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
```
Comment (smaslov-intel) [inline xptifw/CMakeLists.txt:21]: Please add a comment why building with /EHsc

**Review by smaslov-intel — APPROVED** [review]

File: xptifw/CMakeLists.txt (hunk 3 above)
Comment (v-klochkov) [reply xptifw/CMakeLists.txt:21]: It is needed because of the change: "printf" -> "std::cout <<"

Comment (smaslov-intel) [reply xptifw/CMakeLists.txt:21]: can you elaborate even more and add this as a source code comment?

Comment (alexbatashev) [reply xptifw/CMakeLists.txt:21]: Is it ok to do so in separate PR?

Comment (smaslov-intel) [reply xptifw/CMakeLists.txt:21]: Yes, sure. I approved this PR

**Comment by romanovvlad** [comment]
@bader It seems the PR is waiting for your review, could you please do a review?

**Comment by bader** [comment]
> @bader It seems the PR is waiting for your review, could you please do a review?

Sure. For the future, please, [request review via GitHub UI](https://github.com/intel/llvm/pull/2731#issuecomment-762661408) explicitly.

**Review by bader — APPROVED** [review]

**Pull request merged**